packages/venia-concept/src/ServiceWorker/registerRoutes.js

```diff
@@ -83,7 +83,7 @@ export default function() {
      * and the cycle repeats.
      */
     registerRoute(
-        ({ url }) => isHTMLRoute(url),
+        ({ url }) => url.origin === self.location.origin && isHTMLRoute(url),
```

Comment:
Is this needed? The SW only has access to routes under the scope, in this case, the domain that launched it.

Reply:
I don't know about the whole service worker implementation in pwa-studio, but from [here](https://developers.google.com/web/tools/workbox/modules/workbox-routing#how_to_register_a_regular_expression_route) it seems that workbox may have access to external resources. 

And without this change the script was loading through the service worker. The network tab showing it with a gear icon, and the response being the index template.

Reply:
> Is this needed? The SW only has access to routes under the scope, in this case, the domain that launched it.

@revanth0212 I think, based on the link provided, if you pass a regex it will only match against the sw domain, but in our case we are intercepting all urls in the callback function.

Additionally, based on the `isHomeRoute` function it seems like ANY url without sub-paths ie `www.google.com` or `zopim.com` would return `true` so the SW would return _our_ index. And if there was a sub path that happened to match the store code like `www.google.com/en_us` it would _also_ return true and therefor our index file.

So if an external asset (like zen desk's js) is fetched from root, or if a file is requested from a path that is the first level that matches our store code, it will hit this sw route.

```export const isHomeRoute = url => {
    if (url.pathname === '/') {
        return true;
    }

    // If store code is in the url, the home route will be url.com/view_code.
    // A trailing / may or may not follow.
    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
        return AVAILABLE_STORE_VIEWS.some(
            ({ code }) =>
                url.pathname === `/${code}/` || url.pathname === `/${code}`
        );
    }
};```